### PR TITLE
Add small examples for serialization feature

### PR DESCRIPTION
**Pull request by mitmul**
- [ ] Clarify what can be serialized, and what can't.

### CODE REVIEW

File: examples/serialization/load.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import chainer
+import chainer.functions as F
+import chainer.links as L
+
+n_in = 784
+n_units = 100
+n_out = 10
+
+# Network definition
+class MLP(chainer.Chain):
```
Comment (niboshi) [inline examples/serialization/load.py]: How about moving the model definition into a separate file?

File: examples/serialization/load.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __init__(self, n_in, n_units, n_out):
+        super(MLP, self).__init__()
+        with self.init_scope():
```
Comment (niboshi) [inline examples/serialization/load.py]: How about demonstrating `add_persistent`?

**Comment by niboshi** [comment]
Other suggestions:
* User-defined serializable object, not just a link
* Serialization into other formats (e.g. hdf5)

**Comment by niboshi** [comment]
@mitmul Do you have time to work on this PR?

File: examples/serialization/load.py (hunk 1 above)
Comment (mitmul) [reply examples/serialization/load.py]: Thanks. moved

File: examples/serialization/load.py (hunk 2 above)
Comment (mitmul) [reply examples/serialization/load.py]: That's nice idea. I added it.

**Comment by mitmul** [comment]
@niboshi I came back to Chainer development, so restarted working on it. I added some of your suggestions. Could you check it again?

File: examples/serialization/load.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import chainer
+import numpy as np
+import model
+
+
+n_in = 784
+n_units = 100
+n_out = 10
```
Comment (niboshi) [inline examples/serialization/load.py]: How about including these parameters in `model.py`?
We should demonstrate minimal code repetition is required in save/load scripts.

File: examples/serialization/load.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+# Create model object first
```
Comment (niboshi) [inline examples/serialization/load.py]: Logical separation of this script is not clear at first glance.
How about separating into functions, like `load_npz_file` and `load_hdf5_file` (or any better names)?

ditto. for `save` script.

File: examples/serialization/save.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+    # Load the saved HDF5 using h5py
+    print('--- The list of saved params in model.h5 ---')
+    f = h5py.File('model.h5', 'r')
+    for param_key, param in f.items():
+        sys.stdout.write('{}:'.format(param_key))
+        if isinstance(param, h5py.Dataset):
+            print(' {}'.format(param.shape))
+        else:
+            print('')
```
Comment (niboshi) [inline examples/serialization/save.py]: `sys.stdout` and `print` should not be mixed.
How about `print`ing the whole line at once?
```
if ...
    print('{}: {}'.format(param_key, param.shape)
else:
    print('{}:'.format(param_key)
```

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**